<center><img src="https://media-exp1.licdn.com/dms/image/C4E22AQEbIXZiRVkJPQ/feedshare-shrink_2048_1536/0?e=1598486400&v=beta&t=DSWeH9gVxZecLFd10nv-rGomMSJEJP7D2pXl60V1-9A" width="1000"></center>

# SDC Certified Program in Python for Data Science

## Evaluación nivel intermedio

**Materiales para utilizar:** Está permitido el uso de apuntes. Debe usar Python para resolver los ejercicios planteados (incluir el código y programas usados para resolver las preguntas).<br>
**Formato:** el archivo notebook de Jupyter y el pdf debe ser guardado con su apellido paterno y materno sin espacios.<br>
**Asunto:**  Examen_Intermedio_ApellidoPaterno_ApellidoMaterno_Nombres<br>
**Puntaje:** Total: 20 puntos<br>

## Información del conjunto de datos

Los dos conjuntos de datos están relacionados con variantes rojas y blancas del vino portugués "Vinho Verde". . Debido a cuestiones de privacidad y logística, solo están disponibles las variables fisicoquímicas (entradas) y sensoriales (la salida) (por ejemplo, no hay datos sobre tipos de uvas, marcas de vinos, precios de venta de vinos, etc.)

Basado en este conjunto de datos el objetivo es determinar un algoritmo que logre predecir si el vino se categoriza en rojo o blanco haciendo uso de un modelo estadístico de clasificación.


## Información del atributo:

Variables de entrada (basadas en pruebas fisicoquímicas):

1 - fixed acidity(acidez fija)

2 - volatile acidity(acidez volátil)

3 - citric acid(ácido cítrico)

4 - residual sugar(azúcar residual)

5 - chlorides(cloruros)

6 - free sulfur dioxide(dióxido de azufre libre)

7 - total sulfur dioxide(dióxido de azufre total)

8 - density(densidad)

9 - pH

10 - sulphates(sulfatos)

11 - alcohol

Variable de salida (basada en datos sensoriales):

12 - wine_type(white,red)


In [1]:
import pandas as pd
wines = pd.read_csv('D:/Python/2. Nivel II/examen/wines_types.csv')
wines.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6497 entries, 0 to 6496
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         6497 non-null   float64
 1   volatile acidity      6497 non-null   float64
 2   citric acid           6497 non-null   float64
 3   residual sugar        6497 non-null   float64
 4   chlorides             6497 non-null   float64
 5   free sulfur dioxide   6497 non-null   float64
 6   total sulfur dioxide  6497 non-null   float64
 7   density               6497 non-null   float64
 8   pH                    6497 non-null   float64
 9   sulphates             6497 non-null   float64
 10  alcohol               6497 non-null   float64
 11  wine_type             6497 non-null   object 
dtypes: float64(11), object(1)
memory usage: 609.2+ KB


In [3]:
wines.head(10)

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.0              0.17         0.74            12.8      0.045   
1            7.7              0.64         0.21             2.2      0.077   
2            6.8              0.39         0.34             7.4      0.020   
3            6.3              0.28         0.47            11.2      0.040   
4            7.4              0.35         0.20            13.9      0.054   
5            7.2              0.53         0.14             2.1      0.064   
6            7.5              0.27         0.31            17.7      0.051   
7            6.8              0.11         0.27             8.6      0.044   
8            9.0              0.44         0.49             2.4      0.078   
9            7.1              0.23         0.30             2.6      0.034   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 24.0                 126.0  0.99420  3.26       0.38   
1                 32.0                 133.0  0.99560  3.27       0.45   
2                 38.0                 133.0  0.99212  3.18       0.44   
3                 61.0                 183.0  0.99592  3.12       0.51   
4                 63.0                 229.0  0.99888  3.11       0.50   
5                 15.0                  29.0  0.99323  3.35       0.61   
6                 33.0                 173.0  0.99900  3.09       0.64   
7                 45.0                 104.0  0.99454  3.20       0.37   
8                 26.0                 121.0  0.99780  3.23       0.58   
9                 62.0                 148.0  0.99121  3.03       0.56   

   alcohol wine_type  
0     12.2     white  
1      9.9       red  
2     12.0     white  
3      9.5     white  
4      8.9     white  
5     12.1       red  
6     10.2     white  
7      9.9     white  
8      9.2       red  
9     11.3     white

Actividades:

1. Codificar las variable wine_type (white= 1 y red=2) y particionar los datos en entrenamiento (75%) y prueba (25%)
2. Modelar el  tipo de vino (wine_type) en función a las variables predictoras haciendo uso del modelo de regresión logístico. Considerar para las predicciones un punto de corte de 0.6
3. Aplicar los parámetros del modelo de entrenamiento a los datos de testeo, considerar para las predicciones un punto de corte de 0.6
4. Evaluar el modelo de regresión logístico haciendo uso de las métricas de accuracy, sensibilidad. especificidad, auc y curva ROC.
5. Modelar el  tipo de vino (wine_type) en función a las variables predictoras haciendo uso del modelo de análisis discriminante lineal. Considerar para las predicciones un punto de corte de 0.6.
6. Aplicar los parámetros del modelo de entrenamiento a los datos de testeo, considerar para las predicciones un punto de corte de 0.6.
7. Modelar el  tipo de vino (wine_type) en función a las variables predictoras haciendo uso del modelo de análisis discriminante cuadrático. Considerar para las predicciones un punto de corte de 0.6.
8. Aplicar los parámetros del modelo de entrenamiento a los datos de testeo, considerar para las predicciones un punto de corte de 0.6.
9. Evaluar los modelos de análisis discriminante lineal y cuadrático haciendo uso de las métricas de accuracy, sensibilidad. especificidad, auc y curva ROC. 
10. Realice una evaluación comentada de los tres modelos y seleccione uno de ellos (el de mejor performance) y estime la probabilidad de que el tipo de vino sea rojo teniendo en cuenta los siguientes valores para la variables:


- fixed acidity(acidez fija): 7.2
- volatile acidity(acidez volátil): 0.35
- citric acid(ácido cítrico): 0.74
- residual sugar(azúcar residual): 2.2
- chlorides(cloruros): 0.051
- free sulfur dioxide(dióxido de azufre libre): 32.0
- total sulfur dioxide(dióxido de azufre total): 183
- density(densidad): 0.98720
- pH: 3.15
- sulphates(sulfatos): 0.38
- alcohol: 12.0
